internal/mongosh/mongosh.go

```diff
@@ -0,0 +1,53 @@
+// Copyright 2020 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mongosh
+
+import (
+	"os"
+	"os/exec"
+	"runtime"
+	"syscall"
+)
+
+const (
+	mongoShLinux   = "mongosh"
+	mongoShWindows = "mongosh.exe"
+	windows        = "windows"
+)
+
+func FindBinaryInPath() string {
+	binary := mongoShLinux
+	if runtime.GOOS == windows {
+		binary = mongoShWindows
+	}
+
+	if path, err := exec.LookPath(binary); err == nil {
+		return path
+	}
+
+	return ""
+}
+
+func Run(binary, username, password, mongoURI string) error {
+	args := []string{"mongosh", "-u", username, "-p", password, mongoURI}
```

Comment:
[q] wouldn't `binary` include `mongosh`? is it needed as part of the args here?

Reply:
I get this error if I don't use `mongosh`

```
Creating your cluster... [Press 'Ctrl + C' to stop waiting]
Current Mongosh Log ID: 604f3e77427af55a14b5a710
Connecting to:    mongodb://127.0.0.1:27017/asw?directConnection=true
MongoParseError: credentials must be an object with 'username' and 'password' properties
```

Looking at your [PR](https://github.com/mongodb/mongocli/pull/23/files#diff-da48a81e22efdba800404183206d7bc8002fb1fc815d8681343186d212459c13R58), you were doing the same

Reply:
then I guess is ok? I don't even remember what I did back in the day

Reply:
I have looked at the doc but I haven’t seen anything related to this but [here](https://gobyexample.com/execing-processes) says:

> Exec requires arguments in slice form (as opposed to one big string). We’ll give ls a few common arguments. Note that the first argument should be the program name.